liblangutil/Exceptions.h

```diff
@@ -238,35 +248,50 @@ class Error: virtual public util::Exception
 
 	static std::string formatErrorSeverity(Severity _severity)
 	{
-		if (_severity == Severity::Info)
-			return "Info";
-		if (_severity == Severity::Warning)
-			return "Warning";
-		solAssert(isError(_severity), "");
-		return "Error";
+		switch (_severity)
+		{
+		case Severity::Info: return "Info";
+		case Severity::Warning: return "Warning";
+		case Severity::Error: return "Error";
+		}
+		util::unreachable();
 	}
 
-	static std::string formatErrorSeverityLowercase(Severity _severity)
+	static std::string formatErrorType(Type _type)
 	{
-		switch (_severity)
+		switch (_type)
 		{
-		case Severity::Info:
-			return "info";
-		case Severity::Warning:
-			return "warning";
-		case Severity::Error:
-			solAssert(isError(_severity), "");
-			return "error";
+		case Type::IOError: return "IOError";
+		case Type::FatalError: return "FatalError";
+		case Type::JSONError: return "JSONError";
+		case Type::InternalCompilerError: return "InternalCompilerError";
+		case Type::CompilerError: return "CompilerError";
+		case Type::Exception: return "Exception";
+		case Type::CodeGenerationError: return "CodeGenerationError";
+		case Type::DeclarationError: return "DeclarationError";
+		case Type::DocstringParsingError: return "DocstringParsingError";
+		case Type::ParserError: return "ParserError";
+		case Type::SyntaxError: return "SyntaxError";
+		case Type::TypeError: return "TypeError";
+		case Type::UnimplementedFeatureError: return "UnimplementedFeatureError";
+		case Type::YulException: return "YulException";
+		case Type::SMTLogicException: return "SMTLogicException";
+		case Type::Warning: return "Warning";
```

Comment:
Why is "Warning" and "Info" also here? I thought this will then be managed by the severity?

Reply:
Because we don't really have different types of warnings and infos, like we do for errors. Perhaps it would be better to decouple severity from type but that's out of scope of the refactor here. We could discuss this in #13410 instead.